src/index.tsx

```diff
@@ -399,6 +439,7 @@ class App extends React.Component<{}, AppState> {
       drawScene();
       event.preventDefault();
     } else if (event.key === "Backspace") {
+      addOnBeforeUnload();
```

Comment:
This is very error prone to sprinkle those tests at places where we may be changing something. We should figure out the right place to hook this into.

Ideally, what we want is whenever the elements change, to save it. componentDidMount is likely a good candidate.